### PR TITLE
Margrit/socialdistancing

### PR DESCRIPTION
**Pull request by maccxs**
We still adjust the radius of the collider. But we attached a second collider that keeps the former "narrow radius" for collision. If NPC encounters the player or the map limits
we disable the collision with collider1. It will still collide with collider2.
In this way the NPCs keep their distance from each other but not from the walls or the player.
In addition, we adjust the bounciness of collider1, since more social distancing makes us less bouncy. This should also result in a lower velocity for more social distance. More social distancing makes us crave less friction, so we adapt that value as well.
* Potential Problem: We have two circle colliders attached to NPC prefab. I have no idea how those are distinguished.

### CODE REVIEW

File: YouVsVirus/Assets/Scripts/Components/SocialDistancing.cs [hunk 1]
```diff
@@ -7,36 +7,76 @@
...
+    }
+
+    /// <summary>
+    /// If NPC encounters the player or the map limits
+    /// we disable the collision. We have a second collider closer
+    /// to the object with which we can still collide later. In this way
+    /// the NPCs keep their distance from each other but not from the walls
+    /// or the player.
+    /// Potential Problem: We have two circle colliders attached to NPC prefab.
+    /// I have no idea how those are distinguished.
```
Comment (ghost) [inline YouVsVirus/Assets/Scripts/Components/SocialDistancing.cs:68]: @maccxs  That's why I attached a new child game object for the infection radius collider: You can't really distinguish them. You also don't know which of them is returned by `GetComponent<CircleCollider>()`. If you want a specific collider, you'd have to use `GetComponents<CircleCollider>()` which returns a list of colliders, and then examine their properties.

**Review by ghost — COMMENTED** [review]
Looking fine, the NPCs really look like they're avoiding each other. But maybe we can make them move a little slower? Also, we could make only a fraction of the population care for social distancing, depending on the factor.

I still don't like how some NPCs basically become immune to infection through the larger colliders; but for now, this should work fine. Maybe, instead of using larger physical colliders, we could try using a trigger collider as a kind of 'discomfort zone', and make an NPC actively move away from another NPC when they enter this zone.

**Comment by maccxs** [comment]
Hey,

I tried lots of possibilities. I like this one for now. But feel free to try something else out. Do I have to do anything about the colliders now? If so what specifically?  Thanks for the comments!

**Comment by ghost** [comment]
I guess we can leave the colliders as they are; at least I wouldn't know what to change.

**Review by ghost — APPROVED** [review]

**Comment by maccxs** [comment]
Concerning the velocity: I tried that as well. But if they have less velocity the collision has less impulse and the stick together. Then, we would have to change the overall velocity. So far playing the game I find it ok.

**Comment by maccxs** [comment]
@FredyHennig : What if in the OnTriggerEnter2D of the infection, we ask if the actual position +/- spritze size of the human is in the infection radius? Then the "outer collider" would not count. Since the NPCs now have another closer collider, this would trigger as well, when they are really within the infection radius.

**Comment by holke** [comment]
Looks good to me.
I did not have time to look at the code, but think @FredyHennig  already did.

Think we can merge this, however i see two issues that may be changed in the future:
1. Even with 100% social distancing the NPCs get rather close to each other. I think the visual effect should be more here. Can we double the distance?
2. The NPCs do not keep social distance from the player, which i think they should, but seems to be more complicated to implement (especially when the player shoud be able to force break the distance).

**Comment by ghost** [comment]
> 
> 
> @FredyHennig : What if in the OnTriggerEnter2D of the infection, we ask if the actual position +/- spritze size of the human is in the infection radius? Then the "outer collider" would not count. Since the NPCs now have another closer collider, this would trigger as well, when they are really within the infection radius.

@maccxs Sorry, I'm afraid I can't quite follow... Do you intend to increase the size of the trigger zone and add an additional check for distance, and this way solve the invulnerability problem?

**Pull request merged**